doc/sphinx/source/users-guide/index.rst

```diff
@@ -0,0 +1,40 @@
+.. _users-guide-index:
+
+User's Guide
+============
```

Comment:
That's a great idea.  Separately, I was thinking of having an introduction to the language sections while stewing on this the past few days.  I'll put in the start of something for now and refine it as we go.